### PR TITLE
Fix for steer unavailable for RAV4H and Corolla

### PR DESCRIPTION
**Pull request by arne182**
https://github.com/commaai/openpilot/pull/856

### CODE REVIEW

**Comment by geohot** [comment]
I don't understand the implications of this change, can it be written in a more minimal way? Like does the place where apply_toyota_steer_torque_limits is called need to change?

**Comment by ShaneSmiskol** [comment]
I don't think we need the steer angle part, I've tested the code with just the rate limiter and it's worked wonderfully

**Comment by arne182** [comment]
Yes, limits should be applied after the fault not before. Applying it before causing steering faults as the torque rate limit is not held. The only reason to jump down to 0 in one frame is if it is not enabled anymore.

**Comment by arne182** [comment]
Maybe for the Corolla it is only rate that needs to be limited but I (Rav4h) definitely have situations where the rate is much lower but the steering angle goes over 100 and then faults.

**Comment by geohot** [comment]
What does the change look like with "just the rate limiter"?

**Comment by arne182** [comment]
how do I get the ./update_refs.py to run so that this check does not fail.

**Comment by pd0wm** [comment]
We can take care of that before we merge the PR.

File: selfdrive/car/toyota/carcontroller.py [hunk 1]
```diff
@@ -128,20 +128,27 @@ def update(self, enabled, CS, frame, actuators, pcm_cancel_cmd, hud_alert,
...
       self.last_fault_frame = frame
 
     # Cut steering for 2s after fault
-    if not enabled or (frame - self.last_fault_frame < 200):
-      apply_steer = 0
+    if (frame - self.last_fault_frame < 200) or \
+       (abs(CS.angle_steers_rate) > 100 and CS.CP.carFingerprint in (CAR.RAV4H, CAR.COROLLA)) or \
+       (CS.CP.carFingerprint == CAR.RAV4H and abs(CS.angle_steers) > 100):
+      new_steer = 0
       apply_steer_req = 0
```
Comment (pd0wm) [inline selfdrive/car/toyota/carcontroller.py]: Do you want the steer_request bit to be zero while you're winding down?

Comment (arne182) [reply selfdrive/car/toyota/carcontroller.py]: I have tried it without switching it off and then there are still some occurrences where the steering faults. Having it switch to 0 steering request is the right solution.

**Pull request closed without merging**